crates/provider-sdk/src/lib.rs

```diff
@@ -148,31 +150,125 @@ pub struct Context {
 /// The super trait containing all necessary traits for a provider
 pub trait Provider: ProviderHandler + WrpcNats + WrpcDispatch + Send + Sync + 'static {}
 
+/// Configuration of a link that is passed to a provider
+pub trait LinkConfig: Send + Sync {
+    /// Given that the link was established with the source as this provider,
+    /// get the target ID which should be a component
+    fn get_target_id(&self) -> &ComponentId;
```

Comment:
This may or may not be a component ID _technically_, but under the hood it's just a string so it's fine.

Reply:
Ah no, I'll change this to `LatticeTarget` since we have a proper name for it!